webapp/lib/view.dart

```diff
@@ -886,53 +850,29 @@ class TagListView extends BaseView {
     _tagsContainer = new DivElement()
       ..classes.add('tags');
     _tagsActionContainer = new SpanElement()
-      ..classes.add('tags__actions')
-      ..append(
-        new ButtonElement()
-          ..classes.add('button-add-tag')
-          ..text = '+'
-          ..onClick.listen((event) {
-            event.stopPropagation();
-            if (tagsEditable) {
-              var tagElement = new TagView('', model.TagType.Normal, onTagChangedCallback, tagsEditable);
-              _tagsActionContainer.insertAdjacentElement('beforebegin', tagElement.renderElement);
-              tagElement.focus();
-              return;
-            }
-            _addTagDropdown(availableTags, onTagChangedCallback);
-          })
-      );
+      ..classes.add('tags__actions');
+    _tagsActionContainer.append(
+      new AddButtonElement((MouseEvent event) {
+        if (tagsEditable) {
+          var editableTag = new EditableTagElement(onTagChangedCallback);
+          _tagsActionContainer.insertAdjacentElement('beforebegin',editableTag.renderElement);
+          editableTag.focus();
```

Comment:
prefer to use the format:
```
if (...) {
  ...
  return;
}
...
```
rather than `if {} else {}` - especially in very nested code like this, it's quite hard to figure out when the brackets end, so early return is easier to understand.

Reply:
Changed